### DISCUSSION TITLE
How to implement a custom hessian into sqpmethod

### DISCUSSION THREAD

**Discussion by davidhnguyen2000** [casadi/casadi#3538] [Q&A] [answered]
I am trying to use sqpmethod with osqp to solve an optimization control problem with quadratic costs and non-convex constrains. This is leading to a non-convex lagrangian and thus sqp method is failing. Since my costs are quadratic and convex, I would like to use a custom hessian of just the costs rather than the full (non-convex) lagrangian. 

How can I implement this custom hessian into the sqp solver? 

I have been looking into using the "hess_lag" solver option and it seems it requires 4 inputs and a 1 output. What are those 4 inputs? Also, is this the correct route to take for implementing a custom hessain?

Any help is much appreciated. Thank you.

**Comment by jgillis** [comment, marked as answer]
I've amended the FAQ entry here: https://github.com/casadi/casadi/wiki/FAQ:-How-to-specify-a-custom-Hessian-approximation%3F
